packages/graph/hash_graph/lib/graph/src/ontology/types/mod.rs

```diff
@@ -0,0 +1,80 @@
+//! Definitions of the elements of the Type System stored in the [`store`].
+//!
+//! This module contains the definitions of [`DataType`]s, [`PropertyType`]s, [`EntityType`]s, and
+//! [`LinkType`]s. The structs are Rust representations of their meta-schemas defined within the
+//! Block Protocol specification, and are used to validate instances of types using [`serde`]. To
+//! aid with the de/serialization, intermediary structs and helpers are defined within:
+//! - [`uri`] - Definitions of the unique URI identifiers of Types
+//! - [`serde_shared`] - Common intermediary structs that are helpful with de/serialization
+//! - the type modules - specialized structs that live alongside the definitions of the types when
+//!   needed
+//!
+//! [`store`]: crate::store
+use serde::{Deserialize, Serialize};
+use utoipa::Component;
+
+use crate::ontology::{AccountId, VersionId};
+
+pub mod uri;
+
+mod data_type;
+mod entity_type;
+mod link_type;
+mod property_type;
+
+use crate::ontology::types::uri::VersionedUri;
+pub use crate::ontology::types::{
+    data_type::{DataType, DataTypeReference},
+    entity_type::{EntityType, EntityTypeReference},
+    link_type::LinkType,
+    property_type::{PropertyType, PropertyTypeReference},
+};
+
+pub mod error;
+
+mod serde_shared;
```

Comment:
I'm not super happy with this name, but also don't have a suggestion right now. If we cannot come up with a better name I'm fine with leaving it as it is, but I'd prefer not to 🙃

Reply:
I can't think of a better name either, but for what it's worth I called it `serde` because of de/serialization rather than the actual crate

Reply:
cc @hashintel/backend